src/mudlet-lua/lua/Other.lua

```diff
@@ -832,3 +845,51 @@ do
     end
   end
 end
+
+
+-- A utility function based on tempTimers meant to track all sorts of temporary states
+local timeframeTable = {}  -- keeps track of all timeframe timers
+function timeframe(vname, true_time, nil_time, ...)
+  local timerlist = {
+    { 0 }, -- reset variable 'vname' to nil
+    true_time and { true_time, true }, -- when to set variable to true
+    nil_time and { nil_time }, -- when to set variable back to nil
+    ...  -- further list of times and values to set the variable to
+  }
+
+  -- reset any associated timeframes for 'vname'
+  killTimeframe(vname)
+  timeframeTable[vname] = {}
+
+  -- run through timerlist data and create tempTimers to set 'vname' variable data
+  local maxtime = 0
+  for step, data in ipairs(timerlist) do
+    local time = data[1]
+    local value = data[2]
+    maxtime = (time > maxtime) and time or maxtime
+    if time <= 0 then
+      assert(loadstring(format("%s = %s", vname, tostring(value or nil))))()
+    else
+      timeframeTable[vname][step] = tempTimer(time, function()
+        assert(loadstring(format("%s = %s", vname, tostring(value or nil))))()
+      end)
+    end
+  end
+
+  -- final tempTimer to kill the 'vname' timeframe
+  timeframeTable[vname][#timeframeTable[vname] + 1] = tempTimer(maxtime + 0.1, function()
+    killTimeframe(vname)
+  end)
+end
+
+function killTimeframe(vname)
+  if timeframeTable then
+    if timeframeTable[vname] then
```

Comment:
You may want to think about combining both `if`s above to reduce a level of nesting